src/icalendar/cli.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+"""iCalendar utility"""
+from __future__ import unicode_literals
+
+import argparse
+import sys
+from datetime import datetime
+
+from . import Calendar, __version__
+
+
+_template = """Organiser: {organiser}
+Attendees:
+  {attendees}
+Summary: {summary}
+When: {time_from}-{time_to}
+Location: {location}
+Comment: {comment}
+Description:
+
+{description}
+"""
+
+
+def _format_name(address):
+    """Retrieve the e-mail and optionally the name from an address.
+
+    :arg vCalAddress address: An address object.
+
+    :returns str: The name and optionally the e-mail address.
+    """
+    if not address:
+        return ''
+
+    email = address.title().split(':')[1]
+    if 'cn' in address.params:
+        return '{} <{}>'.format(address.params['cn'], email)
+
+    return email
+
+
+def _format_attendees(attendees):
+    """Format the list of attendees.
+
+    :arg any attendees: Either a list, a string or a vCalAddress object.
+
+    :returns str: Formatted list of attendees.
+    """
+    if type(attendees) == list:
+        return '\n  '.join(map(_format_name, attendees))
+    return _format_name(attendees)
+
+
+def _optional(event, key):
+    """Wrapper to prevent a KeyError.
+
+    :arg Event event: An event.
+    :arg str key: Attribute of the event.
+
+    :returns any: The attribute or an empty string.
+    """
+    if key in event:
+        return event[key]
+    return ''
+
+
+def view(input_handle, output_handle):
+    """Make a human readable summary of an iCalendar file.
+
+    :arg stream handle: Open readable handle to an iCalendar file.
+
+    :returns str: Human readable summary.
+    """
+    cal = Calendar.from_ical(input_handle.read())
+
+    for event in cal.walk():
+        if event.name == 'VEVENT':
+            break
+
+    output_handle.write(_template.format(
+        organiser=_format_name(event['organizer']),
```

Comment:
Looks like the organizer is optional. In my case using a google calendar I got:

```Traceback (most recent call last):
  File "/home/pablo2/.virtualenvs/icalendar/bin/icalendar", line 11, in <module>
    load_entry_point('icalendar==4.0.1.dev0', 'console_scripts', 'icalendar')()
  File "/home/pablo2/.virtualenvs/icalendar/lib/python2.7/site-packages/icalendar-4.0.1.dev0-py2.7.egg/icalendar/cli.py", line 118, in main
    args.func(**{k: v for k, v in vars(args).items()
  File "/home/pablo2/.virtualenvs/icalendar/lib/python2.7/site-packages/icalendar-4.0.1.dev0-py2.7.egg/icalendar/cli.py", line 81, in view
    organiser=_format_name(event['organizer']),
  File "/home/pablo2/.virtualenvs/icalendar/lib/python2.7/site-packages/icalendar-4.0.1.dev0-py2.7.egg/icalendar/caselessdict.py", line 42, in __getitem__
    return super(CaselessDict, self).__getitem__(key.upper())
KeyError: u'ORGANIZER'
```

I would remove the `_optional` wrapper and just use `event.get('organizer', '')`

Reply:
Fixed in a18a8e4.